aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,398 @@
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(fromTableDbName)
+          .name(fromTableName)
+          .build());
+      fromTable = response.table();
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchTableException(e, "Cannot rename %s because the table does not exist in Glue", from);
+    }
+    dropTable(from, false);
+    // use the same Glue info to create the new table, pointing to the old metadata
+    TableInput.Builder tableInputBuilder = TableInput.builder()
+        .owner(fromTable.owner())
+        .tableType(fromTable.tableType())
+        .parameters(fromTable.parameters());
+    try {
+      glue.createTable(CreateTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.getDatabaseName(to))
+          .tableInput(tableInputBuilder.name(IcebergToGlueConverter.getTableName(to)).build())
+          .build());
+      LOG.info("Renamed table from {} to {}", from, to);
+    } catch (AwsServiceException | NoSuchTableException e) {
+      // rollback, recreate old table and rethrow exception
+      LOG.error("Fail to rename table from {} to {}, rollback the old table", from, to, e);
+      glue.createTable(CreateTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(fromTableDbName)
+          .tableInput(tableInputBuilder.name(fromTableName).build())
+          .build());
+      throw e;
+    }
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    try {
+      glue.createDatabase(CreateDatabaseRequest.builder()
+          .catalogId(catalogId)
+          .databaseInput(IcebergToGlueConverter.toDatabaseInput(namespace, metadata))
+          .build());
+      LOG.info("Created namespace: {}", namespace);
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+      throw new AlreadyExistsException("Cannot create namespace %s because it already exists in Glue", namespace);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException {
+    if (!namespace.isEmpty()) {
+      // if it is not a list all op, just check if the namespace exists and return itself.
+      if (namespaceExists(namespace)) {
+        return Lists.newArrayList(namespace);
+      }
+      throw new NoSuchNamespaceException(
+          "Glue does not support nested namespace, cannot list namespaces under %s", namespace);
+    }
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<Namespace> results = new ArrayList<>();
+    do {
+      GetDatabasesResponse response = glue.getDatabases(GetDatabasesRequest.builder()
+          .catalogId(catalogId)
+          .nextToken(nextToken)
+          .build());
+      if (response.hasDatabaseList()) {
```

Comment:
I think this needs an `else` to set `nextToken = null` and exit the loop. Otherwise, a response with no database list for a non-null token would cause an infinite loop because the token never changes.